src/query/interpret/eval.hpp

```diff
@@ -479,7 +479,20 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
       case TypedValue::Type::Null:
         return TypedValue(ctx_->memory);
       case TypedValue::Type::Vertex:
-        return TypedValue(GetProperty(expression_result_ptr->ValueVertex(), property_lookup.property_), ctx_->memory);
+        if (property_lookup.evaluation_mode_ == PropertyLookup::EvaluationMode::GET_ALL_PROPERTIES) {
+          if (property_lookup.cache_.size() == 0) {
+            auto props = GetAllProperties(expression_result_ptr->ValueVertex());
+            property_lookup.cache_ = props;
+          }
+
+          if (!property_lookup.cache_.contains(ctx_->properties[property_lookup.property_.ix])) {
+            return TypedValue(ctx_->memory);
+          }
+
+          return TypedValue(property_lookup.cache_.at(ctx_->properties[property_lookup.property_.ix]), ctx_->memory);
```

Comment:
are there multiple property lookups here inside the map literal? I can't see where is it ensured that we share the cache between individual property lookups, maybe I don't see something

Reply:
The `cache_` member of PropertyLookup is the reference to MapLiteral cache (more precisely, the part that stores the cache for the identifier that's in the PropertyLookup). `property_lookup.cache_ = props;` updates the PropertyLookup cache, and that should be visible from inside MapLiteral because of referencing.